ui/src/app/pages/setting/component/adaptertype/adaptertype-list/facility-adapter-list.component.html

```diff
@@ -0,0 +1,551 @@
+
+    <clr-wizard-page [clrWizardPageNextDisabled]="!formPageOne.valid">
+        <ng-template clrPageTitle>Config adapter</ng-template> 
+        <ng-template clrPageNavTitle>
+            Config
+        </ng-template> 
+        <clr-alert [clrAlertType]="'alert-info'" [clrAlertClosable]="false">
+            <div class="alert-item">
+                <ul>
+                    <li style="width: max-content;">This is a commands list contains two predefined commands.</li>
+                    <li>These predefined commands cannot be removed.</li>
+                    <li>If you want to create more commands for your adapter, just use the form below to create.</li>
+                </ul>
+            </div>
+        </clr-alert>
+        <button class="btn btn-primary" style="margin-bottom: -15px;" (click)="openAddCommand()"><clr-icon shape="plus">
+        </clr-icon>Create a new command</button>
+        <clr-datagrid>
+            <clr-dg-column>Name</clr-dg-column>
+            <clr-dg-column>Description</clr-dg-column>
+            <clr-dg-column>Trigger Cycle</clr-dg-column>
+        
+            <clr-dg-row *ngFor="let command of predefineAdapterCommands">
+                <clr-dg-action-overflow>
+                    <button class="action-item"  (click)="onEdit(command)" >Edit</button>
+                    <button class="action-item" [disabled]="checkIsPredefineData(command)" (click)="onDelete(command)" >Delete</button>
+                </clr-dg-action-overflow>
+                <clr-dg-cell>{{command.command}}</clr-dg-cell>
+                <clr-dg-cell title="{{command.description}}" style="max-width: 200px;text-overflow:ellipsis;overflow:hidden;white-space: nowrap;">{{command.description}}</clr-dg-cell>
+                <clr-dg-cell>{{command.triggerCycle}} minutes</clr-dg-cell>
+            </clr-dg-row>
+        </clr-datagrid>
+
+         <form #commandForm="ngForm" *ngIf="onAdd">
+            <section class="form-block">
+                <label>New adapter command</label>
+                <clr-alert [clrAlertType]="'alert-danger'" [clrAlertClosable]="true" [(clrAlertClosed)] = "addCommandErrorClosed">
+                    <div class="alert-item">
+                        <span>The Name is existed.</span>
+                    </div>
+                </clr-alert>
+                <div class="form-group">
+                    <label>Name</label>
+                    <label for="commandname" aria-haspopup="true" role="tooltip"
+                        [class.invalid]="commandname.touched && !commandname.valid"
+                        class="tooltip tooltip-validation tooltip-md tooltip-bottom-right">
+                        <input type="text" id="commandname" required 
+                        pattern="^[a-zA-Z][a-zA-Z0-9_]*$" [(ngModel)]="newCommand.command" name="commandname" #commandname="ngModel">
+                        <span class="tooltip-content">Only letters and numbers can be entered and start with a letter!</span>
```

Comment:
Please use letter and number combination for the adapter name, it must start with a letter.